tests/parser/match_stmt1.py

```diff
@@ -0,0 +1,80 @@
+def _get_slots(cls):
+    match cls.__dict__.get('__slots__'):
+        case None:
+            return
+        case str(slot):
+            yield slot
+        # case iterable if not hasattr(iterable, '__next__'):
+        #     yield from iterable
+        case _:
+            raise TypeError(f"Slots of '{cls.__name__}' cannot be determined")
+
+def check_sequence_then_mapping(x):
+    match x:
+        case [*_]:
+            return "seq"
+        case {}:
+            return "map"
+
+@staticmethod
+def check_mapping_then_sequence(x):
+    match x:
+        case {}:
+            return "map"
+        case [*_]:
+            return "seq"
+
+def file_handler_v1(command):
+    match command.split():
+        case ['show']:
+            print('List all files and directories: ')
+            # code to list files
+        case ['remove',
+              *files]:
```

Comment:
Fixed all the cases, but this is tricky. Here, the parentheses value is not recorded, simply put the `parenlevel = 0`.

Reply:
Why is it not recorded?

Reply:
Ah I see, it's not recorded in the lookahead part. 

Yes, indeed, that's because regex cannot parse nested parentheses.